serialization/serialization_config.go

```diff
@@ -23,75 +23,122 @@ import (
+// SetIdentifiedDataSerializableFactories adds zore or more identified data serializable factories.
+// Identified data serializable factories is a map of factory IDs and corresponding IdentifiedDataSerializable factories.
+func (b *Config) SetIdentifiedDataSerializableFactories(factories ...IdentifiedDataSerializableFactory) {
+	b.identifiedDataSerializableFactories = append(b.identifiedDataSerializableFactories, factories...)
+}
+
+// IdentifiedDataSerializableFactories returns a copy of identified data serializable factories.
+// Identified data serializable factories is a map of factory IDs and corresponding IdentifiedDataSerializable factories.
+func (b *Config) IdentifiedDataSerializableFactories() []IdentifiedDataSerializableFactory {
+	fs := make([]IdentifiedDataSerializableFactory, len(b.identifiedDataSerializableFactories))
+	copy(fs, b.identifiedDataSerializableFactories)
+	return fs
+}
+
+// SetPortableFactories adds zero or more portable factories.
+// Portable factories is a map of factory IDs and corresponding Portable factories.
+func (b *Config) SetPortableFactories(factories ...PortableFactory) {
+	b.portableFactories = append(b.portableFactories, factories...)
 }
 
-// AddPortableFactory adds a portable factory.
-func (b *Config) AddPortableFactory(factory PortableFactory) {
-	b.PortableFactories = append(b.PortableFactories, factory)
+// PortableFactories returns a copy of portable factories
+// portable factories is a map of factory IDs and corresponding Portable factories.
+func (b *Config) PortableFactories() []PortableFactory {
+	fs := make([]PortableFactory, len(b.portableFactories))
+	copy(fs, b.portableFactories)
+	return fs
 }
 
-// AddCustomSerializer adds a customer serializer for the given type.
-func (b *Config) AddCustomSerializer(t reflect.Type, serializer Serializer) error {
+// SetCustomSerializer adds a customer serializer for the given type.
+// custom serializers is a map of object types and corresponding custom serializers.
+func (b *Config) SetCustomSerializer(t reflect.Type, serializer Serializer) error {
+	b.ensureCustomSerializers()
 	if serializer.ID() <= 0 {
 		return errors.New("serializerID must be positive")
 	}
-	b.CustomSerializers[t] = serializer
+	b.customSerializers[t] = serializer
 	return nil
 }
 
-func (b *Config) AddClassDefinition(definition *ClassDefinition) {
-	b.ClassDefinitions = append(b.ClassDefinitions, definition)
+// CustomSerializers returns a copy of custom serializers.
+// custom serializers is a map of object types and corresponding custom serializers.
+func (b *Config) CustomSerializers() map[reflect.Type]Serializer {
+	b.ensureCustomSerializers()
+	sers := map[reflect.Type]Serializer{}
+	for k, v := range b.customSerializers {
+		sers[k] = v
+	}
+	return sers
+}
+
+// SetClassDefinitions adds zore or more class definitions for portable factories.
+func (b *Config) SetClassDefinitions(definitions ...*ClassDefinition) {
```

Comment:
What is the reasoning behind taking references here (`*ClassDefinition`) but struct object above (`factories ...PortableFactory`)

Reply:
`PortableFactory` is an interface, but `ClassDefinition` is a struct.